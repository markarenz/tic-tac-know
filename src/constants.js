// STRINGS
const PLAYER = 'player';
const CPU = 'cpu';
const X = 'x';
const O = 'o';

const AI_LEVELS = {
  AI_LEVEL_RND: 'ai_level_rnd', // Always random
  AI_LEVEL_KI: 'ai_level_ki', // Killer instinct, random until a winning move is found
  AI_LEVEL_HAH: 'ai_level_hah', // Half and half, sometimes SMRT and sometimes KI
  AI_LEVEL_SMRT: 'ai_level_smrt', // Use past games to learn how to win or draw
};

const symbols = [X, O];
const sides = [PLAYER, CPU];
const coinSides = ['heads', 'tails'];
const initOpenCells = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const winningPatterns = [
  [1, 2, 3], // TOP ROW
  [4, 5, 6], // MIDDLE ROW
  [7, 8, 9], // BOTTOM ROW
  [1, 4, 7], // LEFT COL
  [2, 5, 8], // MIDDLE COL
  [3, 6, 9], // RIGHT COL
  [1, 5, 9], // TL-BR DIAG
  [3, 5, 7], // TR-BL DIAG
];

const LOCAL_STORAGE_KEYS = {
  HISTORY: 'tic-tac-know-history',
  WINS_LOSSES: 'tic-tac-know-wins-losses',
};

const whoseTurnMessages = {
  player: 'Your turn',
  cpu: 'CPU Turn',
};
const gameOverMessages = {
  player: 'YOU WIN!',
  cpu: 'CPU WINS',
  draw: 'The Game is a draw.',
};

const APP_STATES = {
  GAME: 'game',
  MENU: 'menu',
};

const MENU_STATUSES = {
  PICK_SIDE: 'pick-side',
  PICK_AI: 'pick-ai',
  START: 'start',
  COMPLETE: 'complete',
};

const sideMessages = {
  x: 'You are playing as X. The CPU is playing as O.',
  o: 'You are playing as O. The CPU is playing as X.',
};
const aiLevelLabels = {
  ai_level_rnd: 'Herp Derp',
  ai_level_ki: 'Killer Instinct',
  ai_level_hah: 'Half & Half',
  ai_level_smrt: 'Mastermind',
};
const aiLevelMessages = {
  ai_level_rnd:
    'Your oppnent is, to put it bluntly, not terribly bright, making moves (and inapropriate noises) at random.',
  ai_level_ki:
    'Your opponent knows enough to be annoyingly good, but just like all your friends from college, it cannot plan ahead.',
  ai_level_hah:
    'Your opponent is playing with one chip tied behind its back, using a mix of strategies, sometimes smart and sometimes not.',
  ai_level_smrt:
    "Your opponent is smarter than you are, more disciplined and, let's face it, better looking. You don't stand a chance.",
};

//
export {
  symbols,
  sides,
  coinSides,
  PLAYER,
  CPU,
  X,
  O,
  whoseTurnMessages,
  gameOverMessages,
  sideMessages,
  aiLevelMessages,
  aiLevelLabels,
  initOpenCells,
  winningPatterns,
  AI_LEVELS,
  LOCAL_STORAGE_KEYS,
  APP_STATES,
  MENU_STATUSES,
};
