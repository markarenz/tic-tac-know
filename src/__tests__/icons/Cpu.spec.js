import { render, screen } from '@testing-library/react';
import { AI_LEVELS } from '../../constants';
import Cpu from '../../icons/Cpu';

describe('Cpu', () => {
  it('renders component - RND', () => {
    const container = render(<Cpu aiLevel={AI_LEVELS.AI_LEVEL_RND} />);
    expect(container).toMatchSnapshot();
  });
  it('renders component - KI', () => {
    const container = render(<Cpu aiLevel={AI_LEVELS.AI_LEVEL_KI} />);
    expect(container).toMatchSnapshot();
  });

  it('renders component - HAH', () => {
    const container = render(<Cpu aiLevel={AI_LEVELS.AI_LEVEL_HAH} />);
    expect(container).toMatchSnapshot();
  });

  it('renders component - SMRT', () => {
    const container = render(<Cpu aiLevel={AI_LEVELS.AI_LEVEL_SMRT} />);
    expect(container).toMatchSnapshot();
  });
});
