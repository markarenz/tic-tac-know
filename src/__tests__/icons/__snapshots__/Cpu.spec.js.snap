// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cpu renders component - HAH 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        version="1.1"
        viewBox="0 0 56.702 45.17"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
          display="inline"
          fill="gray"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M8.226 4.578H48.513V33.593H8.226z"
          display="inline"
          fill="#551b81"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.906"
        />
        <path
          d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
          display="inline"
          fill="#b2b2b2"
          fill-opacity="0.995"
          stroke="#1a1a1a"
          stroke-miterlimit="60.9"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
          display="inline"
          fill="#333"
          fill-opacity="1"
          stroke-width="0.892"
        />
        <circle
          cx="28.387"
          cy="2.547"
          display="inline"
          fill="#330080"
          r="0.75"
          stroke-width="0.845"
        />
        <path
          d="M37.9 22.106v1.713H18.994v-1.713z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
        <path
          d="M22.225 14.692h3.404v3.404h-3.404zm8.499 0h3.383v3.404h-3.383z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      version="1.1"
      viewBox="0 0 56.702 45.17"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
        display="inline"
        fill="gray"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M8.226 4.578H48.513V33.593H8.226z"
        display="inline"
        fill="#551b81"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.906"
      />
      <path
        d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
        display="inline"
        fill="#b2b2b2"
        fill-opacity="0.995"
        stroke="#1a1a1a"
        stroke-miterlimit="60.9"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
        display="inline"
        fill="#333"
        fill-opacity="1"
        stroke-width="0.892"
      />
      <circle
        cx="28.387"
        cy="2.547"
        display="inline"
        fill="#330080"
        r="0.75"
        stroke-width="0.845"
      />
      <path
        d="M37.9 22.106v1.713H18.994v-1.713z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
      <path
        d="M22.225 14.692h3.404v3.404h-3.404zm8.499 0h3.383v3.404h-3.383z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cpu renders component - KI 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        version="1.1"
        viewBox="0 0 56.702 45.17"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
          display="inline"
          fill="gray"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M8.226 4.578H48.513V33.593H8.226z"
          display="inline"
          fill="#551b81"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.906"
        />
        <path
          d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
          display="inline"
          fill="#b2b2b2"
          fill-opacity="0.995"
          stroke="#1a1a1a"
          stroke-miterlimit="60.9"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
          display="inline"
          fill="#333"
          fill-opacity="1"
          stroke-width="0.892"
        />
        <circle
          cx="28.387"
          cy="2.547"
          display="inline"
          fill="#330080"
          r="0.75"
          stroke-width="0.845"
        />
        <path
          d="M29.379 26.6h-1.772v-.148q0-1.602-1.75-1.602h-3.52q-3.1 0-3.501-3.521h1.75q.148 1.77 1.75 1.77h3.521q1.519 0 2.636 1.182Q29.611 23.1 31.13 23.1h3.521q1.624 0 1.75-1.771h1.75q-.4 3.52-3.5 3.52H31.13q-1.75 0-1.75 1.603z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
        <path
          d="M25.57 14.665v3.446h-3.447q-3.055 0-3.447-3.446h1.734q.041 1.671 1.713 1.733v-1.733zm5.16 0h3.426v3.446H30.73z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      version="1.1"
      viewBox="0 0 56.702 45.17"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
        display="inline"
        fill="gray"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M8.226 4.578H48.513V33.593H8.226z"
        display="inline"
        fill="#551b81"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.906"
      />
      <path
        d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
        display="inline"
        fill="#b2b2b2"
        fill-opacity="0.995"
        stroke="#1a1a1a"
        stroke-miterlimit="60.9"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
        display="inline"
        fill="#333"
        fill-opacity="1"
        stroke-width="0.892"
      />
      <circle
        cx="28.387"
        cy="2.547"
        display="inline"
        fill="#330080"
        r="0.75"
        stroke-width="0.845"
      />
      <path
        d="M29.379 26.6h-1.772v-.148q0-1.602-1.75-1.602h-3.52q-3.1 0-3.501-3.521h1.75q.148 1.77 1.75 1.77h3.521q1.519 0 2.636 1.182Q29.611 23.1 31.13 23.1h3.521q1.624 0 1.75-1.771h1.75q-.4 3.52-3.5 3.52H31.13q-1.75 0-1.75 1.603z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
      <path
        d="M25.57 14.665v3.446h-3.447q-3.055 0-3.447-3.446h1.734q.041 1.671 1.713 1.733v-1.733zm5.16 0h3.426v3.446H30.73z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cpu renders component - RND 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        version="1.1"
        viewBox="0 0 56.702 45.17"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
          display="inline"
          fill="gray"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M8.226 4.578H48.513V33.593H8.226z"
          display="inline"
          fill="#551b81"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.906"
        />
        <path
          d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
          display="inline"
          fill="#b2b2b2"
          fill-opacity="0.995"
          stroke="#1a1a1a"
          stroke-miterlimit="60.9"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
          display="inline"
          fill="#333"
          fill-opacity="1"
          stroke-width="0.892"
        />
        <circle
          cx="28.387"
          cy="2.547"
          display="inline"
          fill="#330080"
          r="0.75"
          stroke-width="0.845"
        />
        <path
          d="M20.814 21.51H36.1v6.806q0 1.386-1.019 2.385-.998.998-2.384.998h-1.692q-1.386 0-2.405-.998-.998-.999-.998-2.385v-3.403h-6.787zm8.499 3.403v1.691q0 .714.489 1.203.51.51 1.202.51h1.692q.713 0 1.202-.51.51-.49.51-1.203v-1.691z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
        <path
          d="M22.225 14.692h3.404v3.404h-3.404zm8.499 0h3.383v3.404h-3.383z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      version="1.1"
      viewBox="0 0 56.702 45.17"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
        display="inline"
        fill="gray"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M8.226 4.578H48.513V33.593H8.226z"
        display="inline"
        fill="#551b81"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.906"
      />
      <path
        d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
        display="inline"
        fill="#b2b2b2"
        fill-opacity="0.995"
        stroke="#1a1a1a"
        stroke-miterlimit="60.9"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
        display="inline"
        fill="#333"
        fill-opacity="1"
        stroke-width="0.892"
      />
      <circle
        cx="28.387"
        cy="2.547"
        display="inline"
        fill="#330080"
        r="0.75"
        stroke-width="0.845"
      />
      <path
        d="M20.814 21.51H36.1v6.806q0 1.386-1.019 2.385-.998.998-2.384.998h-1.692q-1.386 0-2.405-.998-.998-.999-.998-2.385v-3.403h-6.787zm8.499 3.403v1.691q0 .714.489 1.203.51.51 1.202.51h1.692q.713 0 1.202-.51.51-.49.51-1.203v-1.691z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
      <path
        d="M22.225 14.692h3.404v3.404h-3.404zm8.499 0h3.383v3.404h-3.383z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cpu renders component - SMRT 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        version="1.1"
        viewBox="0 0 56.702 45.17"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
          display="inline"
          fill="gray"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M8.226 4.578H48.513V33.593H8.226z"
          display="inline"
          fill="#551b81"
          fill-opacity="1"
          stroke="#1a1a1a"
          stroke-opacity="1"
          stroke-width="0.906"
        />
        <path
          d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
          display="inline"
          fill="#b2b2b2"
          fill-opacity="0.995"
          stroke="#1a1a1a"
          stroke-miterlimit="60.9"
          stroke-opacity="1"
          stroke-width="0.964"
        />
        <path
          d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
          display="inline"
          fill="#333"
          fill-opacity="1"
          stroke-width="0.892"
        />
        <circle
          cx="28.387"
          cy="2.547"
          display="inline"
          fill="#330080"
          r="0.75"
          stroke-width="0.845"
        />
        <path
          d="M38.026 21.165v.145q0 1.885-1.574 3.46-1.575 1.574-4.29 1.574h-7.251q-2.714 0-4.289-1.574-1.575-1.575-1.575-3.46v-.145h1.72q.207 1.74 1.72 1.74h12.1q1.533 0 1.72-1.74z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
        <path
          d="M25.57 14.665v3.446h-3.447q-3.055 0-3.447-3.446h1.734q.041 1.671 1.713 1.733v-1.733zm5.16 0h3.426v3.446H30.73z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
        <path
          d="M28.247 13.524l-8.373-5.351V6.165l8.373 5.35 8.373-5.35v2.008z"
          display="inline"
          fill="#fff"
          stroke="#1a1a1a"
          stroke-opacity="0"
          stroke-width="0"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      version="1.1"
      viewBox="0 0 56.702 45.17"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.245.54h36.25c3.47 0 5.246 1.82 5.246 4.081v30.85c0 2.261-1.448 4.081-3.245 4.081H8.245c-1.798 0-3.246-1.82-3.246-4.08V4.622C5 2.36 6.775.54 10.245.54z"
        display="inline"
        fill="gray"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M8.226 4.578H48.513V33.593H8.226z"
        display="inline"
        fill="#551b81"
        fill-opacity="1"
        stroke="#1a1a1a"
        stroke-opacity="1"
        stroke-width="0.906"
      />
      <path
        d="M.502 39.92v1.47c0 1.787 1.155 3.24 2.61 3.24h50.49c1.455 0 2.636-1.453 2.636-3.24v-1.47z"
        display="inline"
        fill="#b2b2b2"
        fill-opacity="0.995"
        stroke="#1a1a1a"
        stroke-miterlimit="60.9"
        stroke-opacity="1"
        stroke-width="0.964"
      />
      <path
        d="M0 39.703l4.487-3.635h47.524l4.691 3.51z"
        display="inline"
        fill="#333"
        fill-opacity="1"
        stroke-width="0.892"
      />
      <circle
        cx="28.387"
        cy="2.547"
        display="inline"
        fill="#330080"
        r="0.75"
        stroke-width="0.845"
      />
      <path
        d="M38.026 21.165v.145q0 1.885-1.574 3.46-1.575 1.574-4.29 1.574h-7.251q-2.714 0-4.289-1.574-1.575-1.575-1.575-3.46v-.145h1.72q.207 1.74 1.72 1.74h12.1q1.533 0 1.72-1.74z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
      <path
        d="M25.57 14.665v3.446h-3.447q-3.055 0-3.447-3.446h1.734q.041 1.671 1.713 1.733v-1.733zm5.16 0h3.426v3.446H30.73z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
      <path
        d="M28.247 13.524l-8.373-5.351V6.165l8.373 5.35 8.373-5.35v2.008z"
        display="inline"
        fill="#fff"
        stroke="#1a1a1a"
        stroke-opacity="0"
        stroke-width="0"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
